### PR TITLE
e2e tst: add actions to create github org

### PR DESCRIPTION
**Pull request by burmudar**
Add actions to create github org and also to tear it down

## Test plan
tested in wb/e2e/poc

### CODE REVIEW

**Comment by burmudar** [comment]
Current dependencies on/for this PR:
* main
  * **PR #56774** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56774" target="_blank">Graphite</a> 
    * **PR #56775** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56775" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56775?utm_source=stack-comment).

**Review by jhchabran — CHANGES_REQUESTED** [review]

File: dev/tst/github_client.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		io.Copy(os.Stdout, resp.Body)
+		return nil, errors.Newf("failed to create org %q - GitHub status code %d: %v", name, resp.StatusCode, err)
+	}
+	return org, err
+}
+
+func (gh *GitHubClient) UpdateOrg(ctx context.Context, org *github.Organization) (*github.Organization, error) {
+	_, resp, err := gh.c.Organizations.Edit(ctx, org.GetLogin(), org)
+	if resp.StatusCode >= 400 {
+		io.Copy(os.Stdout, resp.Body)
```
[the remark this answers is not in the document]
Comment (burmudar) [reply dev/tst/github_client.go:58]: Ah good catch!

[the remark this answers is not in the document]
Comment (burmudar) [reply dev/tst/github_client.go]: Fixed

File: dev/codehost_testing/org.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+// Get returns the corresponding GitHub Organization object that was created by the `CreateOrg`
+//
+// This method will only return a Org if the Scenario that created it has been applied otherwise
+// it will panic.
+func (o *Org) Get(ctx context.Context) (*github.Organization, error) {
+	if o.s.IsApplied() {
+		return o.s.client.GetOrg(ctx, o.name)
+	}
+	panic("cannot retrieve org before scenario is applied")
```
Comment (jhchabran) [inline dev/codehost_testing/org.go:29]: What about doing a `t.Fatal` instead, as we can get `t` through the scenario.

File: dev/codehost_testing/org.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+func (o *Org) get(ctx context.Context) (*github.Organization, error) {
+	return o.s.client.GetOrg(ctx, o.name)
+}
+
+// AllowPrivateForks adds an action to the scenario to enable private forks and repos for the org
+func (o *Org) AllowPrivateForks() {
+	updateOrgPermissions := &action{
+		name: "org:permissions:update:" + o.name,
+		apply: func(ctx context.Context) error {
+
```
Comment (jhchabran) [inline dev/codehost_testing/org.go:43]:
```suggestion
```

File: dev/codehost_testing/scenario.go [hunk 4]
```diff
@@ -211,3 +214,32 @@ func (s *GithubScenario) Teardown(ctx context.Context) error {
...
+			if err != nil {
+				return err
+			}
+			baseOrg.name = org.GetLogin()
+			return nil
+		},
+		teardown: func(context.Context) error {
+			host := baseOrg.s.client.cfg.URL
+			deleteURL := fmt.Sprintf("%s/organizations/%s/settings/profile", host, baseOrg.name)
+			fmt.Printf("Visit %q to delete the org\n", deleteURL)
```
Comment (jhchabran) [inline dev/codehost_testing/scenario.go:238]: I was about to suggest using `t.Log` but that would have the unfortunate consequence of hiding this message if we're not running tests in verbose mode.

File: dev/codehost_testing/org.go (hunk 2 above)
Comment (burmudar) [reply dev/codehost_testing/org.go:29]: Yeah I think that would be much better

**Review by jhchabran — APPROVED** [review]

**Pull request merged**